Source/CBLSyncConnection+Pull.m

```diff
@@ -162,11 +164,17 @@ - (void) handleIncomingRevision: (BLIPRequest*)request {
     NSDictionary* attachments = nil;
     NSString* docID;
     NSData* json = request.body;
-    if (memmem(json.bytes, json.length, "\"_attachments\":", 15) != NULL) {
+    if (self.onSyncAccessCheck && memmem(json.bytes, json.length, "\"_attachments\":", 15) != NULL) {
```

Comment:
Why is `self.onSyncAccessCheck` added there? This test checks for whether there could be attachments in the doc; it doesn't depend on whether there's an access check registered.